File: 2020/01/part1.js
const fs = require("fs");

const input = fs.readFileSync("input.txt", "utf-8").split("\n").map(line => parseInt(line, 10));

for (let i = 0; i < input.length; i++)
    for (let j = i + 1; j < input.length; j++)
        if (input[i] + input[j] == 2020)
            console.log(input[i] * input[j]);

File: 2022/01/part1.rs
#!/usr/bin/env rust-script

impl<T> Pipe for T {}
trait Pipe: Sized {
	fn pipe<B, F>(self, f: F) -> B
	where
		F: FnOnce(Self) -> B,
	{
		f(self)
	}
}

std::fs::read_to_string("input.txt")
	.unwrap()
	.split("\n\n")
	.map(|record| {
		record
			.lines()
			.map(|line| line.parse::<u32>().unwrap())
			.sum::<u32>()
	})
	.max()
	.unwrap()
	.pipe(|answer| println!("{answer}"));
